font-codegen/src/record.rs

```diff
@@ -79,20 +79,33 @@ fn generate_read_with_args(item: &Record) -> TokenStream {
     let args = item.attrs.read_args.as_ref().unwrap();
     let args_type = args.args_type();
     let destructure_pattern = args.destructure_pattern();
-    let field_size_expr = item.fields.iter().map(Field::record_len_expr);
+    let field_size_expr: Vec<_> = item.fields.iter().map(Field::record_len_expr).collect();
     let field_inits = item.fields.iter().map(Field::record_init_stmt);
     let constructor_args = args.constructor_args();
     let args_from_constructor_args = args.read_args_from_constructor_args();
 
+    let comp_size_expr = match field_size_expr.as_slice() {
+        [] => panic!("should never be empty"),
+        [one_expr] => quote!( Ok(#one_expr) ),
+        exprs => {
+            quote! {
+                let mut result = 0usize;
+                #( result = result.checked_add(#exprs).ok_or(ReadError::OutOfBounds)?; )*
```

Comment:
👍  Looking at this change, indeed, I think _this_ is the spot I wasn't sure about how to rephrase due to the repetition operator at the end. Thanks.